File: corpus/java/constructor_calls/.repo/app/svc/Maker.java
package app.svc;

import app.model.Base;
import app.model.Box;
import app.model.Listener;
import app.model.Order;

public class Maker {
    public void plain() {
        new Order();
    }

    public void withArgs() {
        Order o = new Order(7);
    }

    public void qualified() {
        Object o = new app.model.Order();
    }

    public void nested() {
        Order.Line line = new Order.Line();
    }

    public void generic() {
        Box<Order> b = new Box<Order>();
    }

    public void diamond() {
        Box<Order> b = new Box<>();
    }

    public void anonymousInterface() {
        Listener l = new Listener() {
            public void fired() {
                new Order();
            }
        };
    }

    public void anonymousClass() {
        Base b = new Base() {
            public void run() {
            }
        };
    }

    public Order returned() {
        return new Order();
    }

    public void array() {
        Order[] orders = new Order[3];
    }

    public void methodReference() {
        java.util.function.Supplier<Order> make = Order::new;
    }

    public void externalGeneric() {
        java.util.List<Order> orders = new java.util.ArrayList<Order>();
    }

    public void localClass() {
        class Tmp {
            Order go() {
                return new Order();
            }
        }
    }
}

File: corpus/csharp/constructor_calls/.repo/App/Svc/Maker.cs
using System.Collections.Generic;
using App.Model;
using App.Util;
using Timer = System.Threading.Timer;

namespace App.Svc
{
    public class Maker
    {
        public void Plain() { var o = new Order(); }

        public void WithArgs() { Order o = new Order(7); }

        public void Qualified() { var o = new App.Model.Order(); }

        public void Nested() { var line = new Order.Line(); }

        public void Generic() { var b = new Box<Order>(); }

        public void Initializer() { var o = new Order { Id = 3 }; }

        public void TargetTyped() { Order o = new(); }

        public void Struct() { var p = new Point(1); }

        public void Record() { var m = new Money(2m); }

        public Order Returned() => new Order();

        public Order TargetTypedReturn() { return new(); }

        public void Array() { var a = new Order[3]; }

        public void ExternalGeneric() { var l = new List<Order>(); }

        public void Shadow<Order>() where Order : new() { var x = new Order(); }

        public void Aliased() { var t = new Timer(null); }
    }
}
